### PR TITLE
add and use the `prefix` option in http frontend util to fix localization

### PR DESCRIPTION
**Pull request by ETLaurent**
fixes localizing issue introduced in #4112

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,11 @@
 are made before the last `apostrophe:modulesRegistered` handler has fired.
 If you need to call Apostrophe's `find()` methods at startup,
 it is best to wait for the `@apostrophecms/doc:beforeReplicate` event.
+* Allow `@` when a piece is a template and `/@` for page templates (doc-template-library module).
```
Comment (ETLaurent) [inline CHANGELOG.md:12]: @falkodev like the entry below, this line was merged into a released section, where it should have been merged into UNRELEASED

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,11 @@
...
+
+### Fixes
+
+* Fix 404 error when viewing/editing a doc which draft has a different version of the slug than the published one.
```
Comment (ETLaurent) [inline CHANGELOG.md:16]: same comment as above

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue [hunk 3]
```diff
@@ -538,11 +538,11 @@ export default {
       const doc = await apos.http.get(`${action}/${this.context.aposDocId}`, {
         qs: {
           aposMode: this.draftMode,
-          project: { _url: 1 }
+          project: { slug: 1 }
         }
       });
 
-      const url = apos.http.addQueryToUrl(doc._url, qs);
+      const url = apos.http.addQueryToUrl(doc.slug, qs);
```
Comment (ETLaurent) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue:545]: Unlike `slug`, `_url`  contains the locale prefix, which results in having the final url constructed with a double locale prefix such as `/fr/fr/some-page`.

cc @haroun

**Comment by ETLaurent** [comment]
Running the whole visual regression tests: https://github.com/apostrophecms/testbed/actions/runs/5136545143

**Review by boutell — CHANGES_REQUESTED** [review]
Slugs are not URLs. Piece slugs don't even have a `/`. Slugs in general can never be trusted to be the URL in Apostrophe. We can't do it this way.

I recognize there is a problem we do have to find a fix for, but I am not sure where the doubled prefix is coming from. Is it coming from `apos.http` somehow?

**Comment by ETLaurent** [comment]
> Slugs are not URLs. Piece slugs don't even have a `/`. Slugs in general can never be trusted to be the URL in Apostrophe. We can't do it this way.

I see, but somehow it works for all pages and piece pages...

> I recognize there is a problem we do have to find a fix for, but I am not sure where the doubled prefix is coming from. Is it coming from `apos.http` somehow?

apos.http might be adding the prefix somewhere, and even the piece page url, looking were it is done...

**Comment by ETLaurent** [comment]
the locale prefix is added here: https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/util/ui/src/http.js#L111

**Comment by ETLaurent** [comment]
newer run after latest core changes: https://github.com/apostrophecms/testbed/actions/runs/5137171642

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,7 @@ are made before the last `apostrophe:modulesRegistered` handler has fired.
 If you need to call Apostrophe's `find()` methods at startup,
 it is best to wait for the `@apostrophecms/doc:beforeReplicate` event.
 * Allow `@` when a piece is a template and `/@` for page templates (doc-template-library module).
+* Adds a `doNotAddPrefix` option to frontend http util module in order to avoid adding a prefix to already-prefixed URLs when making frontend calls.
```
Comment (boutell) [inline CHANGELOG.md:13]: Go with `prefix: false` instead. This is more consistent with our pattern elsewhere. We avoid options with negated names.

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue [hunk 5]
```diff
@@ -538,18 +538,19 @@ export default {
...
+      const url = apos.http.addQueryToUrl(doc._url, qs);
       const content = await apos.http.get(url, {
         qs,
         headers: {
           'Cache-Control': 'no-cache'
         },
         draft: true,
-        busy: true
+        busy: true,
+        doNotAddPrefix: true
```
Comment (boutell) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue:553]: `prefix: false`

File: modules/@apostrophecms/util/ui/src/http.js [hunk 6]
```diff
@@ -104,9 +106,8 @@ export default () => {
       });
     }
 
-    if (apos.prefix) {
-      // We don't need a prefix if the target URL is already prefixed,
-      // which any absolute URL should be
+    if (apos.prefix && !options.doNotAddPrefix) {
```
Comment (boutell) [inline modules/@apostrophecms/util/ui/src/http.js:109]: `options.prefix !== false`

**Review by boutell — CHANGES_REQUESTED** [review]
Right strategy, note on avoiding negated names

**Review by boutell — APPROVED** [review]
Looks good. Please send Bob an `a3-docs` PR.

**Comment by ETLaurent** [comment]
newer tests run: https://github.com/apostrophecms/testbed/actions/runs/5137900205

**Pull request merged**